yarn-project/end-to-end/src/e2e_cross_chain_messaging.test.ts

```diff
@@ -230,25 +213,21 @@ describe('e2e_cross_chain_messaging', () => {
     const [secretForL2MessageConsumption, secretHashForL2MessageConsumption] =
       crossChainTestHarness.generateClaimSecret();
 
-    const entryKey = await crossChainTestHarness.sendTokensToPortalPrivate(
+    const msgLeaf = await crossChainTestHarness.sendTokensToPortalPrivate(
       Fr.random(),
       bridgeAmount,
       secretHashForL2MessageConsumption,
     );
     expect(await crossChainTestHarness.getL1BalanceOf(ethAccount)).toBe(0n);
-    expect(await crossChainTestHarness.inbox.read.contains([entryKey.toString()])).toBeTruthy();
-
-    // Wait for the archiver to process the message
-    await delay(5000); /// waiting 5 seconds.
 
-    // Perform an unrelated transaction on L2 to progress the rollup. Here we mint public tokens.
-    await crossChainTestHarness.mintTokensPublicOnL2(0n);
+    // Wait for the message to be available for consumption
+    await crossChainTestHarness.makeMessageConsumable(msgLeaf);
 
     const content = Fr.fromBufferReduce(
       sha256(
         Buffer.concat([
-          keccak(Buffer.from('mint_public(bytes32,uint256,address)')).subarray(0, 4),
-          serializeToBuffer(...[ownerAddress, new Fr(bridgeAmount), ethAccount.toBuffer32()]),
+          keccak(Buffer.from('mint_public(bytes32,uint256)')).subarray(0, 4),
```

Comment:
For consistency, this one should probably also use the functionselector that you used higher up?

Reply:
Addressed in 54841bdd7847f103b273ff58800b43ea5115e78d